src/app/Library/CrudPanel/Traits/Read.php

```diff
@@ -82,8 +84,29 @@ public function autoEagerLoadRelationshipColumns()
     {
         $relationships = $this->getColumnsRelationships();
 
-        if (count($relationships)) {
-            $this->with($relationships);
+        foreach ($relationships as $relation) {
+            if (strpos($relation, '.') !== false) {
+                $relation_parts = explode('.', $relation);
+                $last_relation_part = array_pop($relation_parts);
+
+                $last_valid_relation_method = array_reduce(array_splice($relation_parts, 0, count($relation_parts)), function ($obj, $method) {
+                    try {
+                        $result = $obj->$method();
+
+                        return $result->getRelated();
+                    } catch (Exception $e) {
+                        return $method;
+                    }
+                }, $this->model);
+
+                // when this keys don't match means the last part of the relation string is the attribute in the relation and
+                // not a nested relation. In that case, we should eager load the relation but not the attribute
+                if ($last_valid_relation_method != $last_relation_part) {
```

Comment:
Also, since you first `array_pop($relation_parts);`, `$relation_parts` will no longer have the last part right?
So this condition will always be true. Am I wrong? 😅

Reply:
Right, this should be `end()` (we don't care about array pointer).